crates/solver/src/liquidity/zeroex.rs

```diff
@@ -0,0 +1,74 @@
+use super::SettlementHandling;
+use crate::interactions::ZeroExInteraction;
+use crate::liquidity::{Exchange, LimitOrder, Liquidity};
+use crate::settlement::SettlementEncoder;
+use anyhow::Result;
+use contracts::IZeroEx;
+use model::order::OrderKind;
+use primitive_types::U256;
+use shared::zeroex_api::{Order, OrderRecord, OrdersQuery, ZeroExApi};
+use std::sync::Arc;
+
+pub struct ZeroExLiquidity {
+    pub api: Arc<dyn ZeroExApi>,
+    pub zeroex: IZeroEx,
+}
+
+impl ZeroExLiquidity {
+    pub async fn get_liquidity(&self) -> Result<Vec<Liquidity>> {
```

Comment:
I think we should filter orders that aren't in the set of `user_order_tokens ∪ base_tokens`. This would be consistent with what we do for other liquidity sources like Uniswap and Balancer.

I think this can go in a follow up.

Reply:
I mean I can do that but I'm not sure what the purpose would be.
If quasimodo is able to handle all the orders wouldn't it be better to give it all the orders we have?
Same for Uniswap and Balancer.

Reply:
Maybe this is something that @marcovc can weigh-in on. Agree that if the solver can handle it than we should give it to the solver.

> Same for Uniswap

<details><summary>I think that would be 10s of thousands of pools.</summary>

```js
let count = 0;
let current = ethers.constants.AddressZero;
do {
  const { data: { pairs } } = await graphQuery(
    "uniswap/uniswap-v2",
    `{
      pairs(
        first: 1000,
        where: {
          id_gt: "${current}"
        }
      ) {
        id
      }
    }`,
  );
  count += pairs.length;
  current = pairs.pop()?.id;
} while (current);

console.log(`Final count: ${count}`);
```

I ran this and got `67538` pools.

</details>